apps/api/src/app/groups/groups.controller.ts

```diff
@@ -365,6 +377,52 @@ export class GroupsController {
         throw new NotImplementedException()
     }
 
+    @Post("/member/:member")
+    @ApiBody({ type: AddMemberToGroupsDto })
+    @ApiHeader({ name: "x-api-key", required: true })
+    @ApiCreatedResponse({ isArray: true, type: Group })
+    @ApiOperation({
+        description:
+            "Adds a member to multiple groups. Requires an API Key in the headers or a valid session."
+    })
+    async addMemberToGroups(
+        @Param("member") memberId: string,
+        @Body() dto: AddMemberToGroupsDto,
+        @Headers() headers: Headers,
+        @Req() req: Request
+    ) {
```

Comment:
```suggestion
    ): Promise<void> {
```

What about returning something like this, similar to the other functions related to adding members? This way the following code is not needed in this function:

```ts
for await (const group of groups) {
            const fingerprint = await this.groupsService.getFingerprint(
                group.id
            )

            groupsToResponseDTO.push(mapGroupToResponseDTO(group, fingerprint))
        }

        return groupsToResponseDTO
```

The `groups.service.ts` functions don't need to change, it's fine to return the groups there. 

The idea is to try to avoid sending a lot of data to the frontend, like all the group info with the list of members.

What do you think?

Reply:
Makes sense, will update.